matching-service/model/socket-room-orm.js

```diff
@@ -23,16 +23,34 @@ export const ormFindOneRoom = async (filter) => {
     return { err }
   }
 }
-export const ormCreateRoom = async (room) => {
+export const ormCreateRoom = async (room, res) => {
   try {
-    return await createRoom(room)
+    const roomModel = await createRoom(room)
+    const newRoom = roomModel.save((err) => {
+      if (err) {
+        // console.log(err)
+        if (err.name === 'MongoServerError' && err.code === 11000) {
+          // Duplicate room_id
+          return res
+            .status(409)
+            .send({ success: false, message: 'Room already exist!' })
+        } else {
+          // Some other error
+          return res.status(422).send({ success: false, message: err })
+        }
+      } else {
+        return res
+          .status(201)
+          .send({ success: true, message: 'Room created successfully!' })
+      }
+    })
```

Comment:
I'm so so sorry but as I mentioned in my previous comment, the error response logic can remain in the controller. There shouldn't be any HTTP response handling at this ORM layer of the application (it should remain in the controller layer) 😬

Reply:
fixed!